src/main/Main.java

```diff
@@ -1,7 +1,316 @@
 package main;
 
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import main.Enums.AccessLevel;
+import main.Enums.ActionResult;
+import utils.FileManager;
+
+import java.util.Date;
+import java.util.Scanner;
+import java.io.*;
+import java.util.stream.Collectors;
+
 public class Main {
+
+
+
+    public static boolean repeatFlag = true;
+    public static int repeatFlagCounter = 0;
+
+    public static String currentTime(){
+
+        LocalDateTime myDateObj = LocalDateTime.now();
+        DateTimeFormatter myFormatObj = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss");
+        String formattedDate = myDateObj.format(myFormatObj);
+        return formattedDate;
+
+    }
+
+
+    public static AccessLevel login(){
+
+        String result = "not_ok";
+        String accessLevel = "";
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please enter your username and password");
+        String username = scanner.nextLine();
+        String password = scanner.nextLine();
+        StringBuilder allString = new StringBuilder();
+
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new FileReader(
+                    "src/resources/Usernames-and-Passwords"));
+            String line = reader.readLine();
+
+
+            while (line != null) {
+               // System.out.println(line);
+                // read next line
+
+                String ins1 = line;
+                String[] ins2 = ins1.split(",");
+                String username2 = ins2[3];
+                String password2 = ins2[4];
+
+                if (username.equals(username2) && password.equals(password2)) {
+
+
+                    result = "ok";
+                    accessLevel = ins2[5];
+                    //System.out.println("Success");
+                    //System.out.println(manager2[7]);
+                    ins2[7] = currentTime();
+                    StringBuilder ins3 = new StringBuilder();
+                    for (int i = 0; i < 8; i++) {
+                        if (i != 7) {
+                            ins3.append(ins2[i] + ",");
+
+                       } else {
+                            ins3.append(ins2[i]);
+                        }
+
+                    }
+
+                   allString.append(ins3);
+                    allString.append("\n");
+
+                }
+
+                else{
+                    allString.append(line);
+                    allString.append("\n");
+                }
+
+
+
+
+                line = reader.readLine();
+            }
+
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+
+       FileManager fmanager = new FileManager("src/resources/Usernames-and-Passwords");
+        fmanager.write(allString.toString(), false);
+
+        if(result == "ok"){
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.